sentence_transformers/losses/MatryoshkaLoss.py

```diff
@@ -21,6 +21,11 @@ def set_dim(self, dim):
         self.idx = 0
 
     def shrink(self, tensor: Tensor) -> Tensor:
+        tensor_dim = tensor.shape[-1]
+        if self.dim > tensor_dim:
+            raise ValueError(
+                f"Dimension {self.dim} in matryoshka_dims is greater than the model's embedding dimension: {tensor_dim}"
```

Comment:
```suggestion
                f"Dimension {self.dim} in matryoshka_dims cannot be greater than the model's embedding dimension: {tensor_dim}"
```
Perhaps this is more descriptive to notify the user what they are doing wrong?

And I'm fine with it being a `ValueError` rather than a warning.